src/main/groovy/io/sentry/android/gradle/GenerateSentryProguardUuidTask.java

```diff
@@ -0,0 +1,54 @@
+package io.sentry.android.gradle;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.UUID;
+
+abstract class GenerateSentryProguardUuidTask extends DefaultTask {
+    @OutputDirectory
+    public abstract DirectoryProperty getOutputDirectory();
+
+    @Internal
+    public Provider<UUID> getOutputUuid() {
+        return getOutputUuidInternal();
+    }
+
+    @Internal
+    protected abstract Property<UUID> getOutputUuidInternal();
+
+    @Internal
+    private final Provider<RegularFile> outputFile = getOutputDirectory().map(dir -> dir.file("sentry-debug-meta.properties"));
+
+    public GenerateSentryProguardUuidTask() {
+        getOutputs().upToDateWhen(spec -> false);
```

Comment:
does it mean each build would generate a new UUID?

Reply:
Yeah, exactly. By default Gradle does not consider tasks up to date. However, if you add any `@Input[Directory|File]` or `@Output[Directory|File]` then Gradle will assume your task is up to date if the input and output is the same as in the last run. For this task we always want it to run when a new build is performed so we effectively "disable" the up to date checks.